src/Picasso_ParticleInterpolation.hpp

```diff
@@ -279,6 +281,323 @@ KOKKOS_INLINE_FUNCTION void divergence( const SplineDataType& sd,
 } // end namespace P2G
 
 //---------------------------------------------------------------------------//
+template <int InterpolationOrder, class ParticleFieldType, class OldFieldType,
+          class InterpolationType>
+struct Particle2Grid;
+
+//---------------------------------------------------------------------------//
+// Project particle enthalpy/momentum to grid. PolyPIC variant
+//---------------------------------------------------------------------------//
+template <int InterpolationOrder, class ParticleFieldType, class OldFieldType>
+struct Particle2Grid<InterpolationOrder, ParticleFieldType, OldFieldType,
+                     PolyPicTag>
+{
+    // Explicit time step.
+    double dt;
+
+    template <class LocalMeshType, class GatherDependencies,
+              class ScatterDependencies, class LocalDependencies,
+              class ParticleViewType>
+    KOKKOS_INLINE_FUNCTION void
+    operator()( const LocalMeshType& local_mesh, const GatherDependencies&,
+                const ScatterDependencies& scatter_deps,
+                const LocalDependencies&, ParticleViewType& particle ) const
+    {
+        // Get particle data.
+        auto f_p = Picasso::get( particle, ParticleFieldType() );
+        auto v_p = Picasso::get( particle, PolyPIC::Field::Velocity() );
+        auto m_p = Picasso::get( particle, Example::Mass() );
```

Comment:
Need to template

Reply:
any reason why need to template `Mass`?

Reply:
You're right, we could leave it (but replace the namespace with Picasso::Field::Mass instead)